uaclient/jobs/tests/test_gcp_auto_attach.py

```diff
@@ -41,20 +59,88 @@ def test_gcp_auto_attach(
             assert (
                 "Disabling gcp_auto_attach job. Not running on GCP instance"
             ) in caplog_text()
-
         else:
             assert cfg.gcp_auto_attach_timer == 1000
             assert m_write.call_count == 1
             assert m_auto_attach.call_count == 1
 
     @mock.patch("uaclient.jobs.gcp_auto_attach.get_cloud_type")
+    @mock.patch.object(UAAutoAttachGCPInstance, "get_licenses_from_identity")
+    @mock.patch("uaclient.jobs.gcp_auto_attach.get_platform_info")
     @mock.patch("uaclient.jobs.gcp_auto_attach.action_auto_attach")
     def test_gcp_job_dont_fail_if_non_auto_attach_image_error_is_raised(
-        self, m_auto_attach, m_cloud_type, FakeConfig
+        self,
+        m_auto_attach,
+        m_platform_info,
+        m_get_licenses,
+        m_cloud_type,
+        FakeConfig,
     ):
         m_cloud_type.return_value = ("gce", None)
+        m_get_licenses.return_value = ["test-license-id"]
+        m_platform_info.return_value = {"series": "ubuntu-lts"}
         m_auto_attach.side_effect = NonAutoAttachImageError("error")
         cfg = FakeConfig()
 
         assert gcp_auto_attach(cfg) is None
         assert m_auto_attach.call_count == 1
+
+    @mock.patch("uaclient.jobs.gcp_auto_attach.get_cloud_type")
+    @mock.patch.object(UAAutoAttachGCPInstance, "get_licenses_from_identity")
+    @mock.patch("uaclient.jobs.gcp_auto_attach.get_platform_info")
+    @mock.patch("uaclient.jobs.gcp_auto_attach.action_auto_attach")
+    def test_gcp_job_dont_fail_if_licenses_fail(
+        self,
+        m_auto_attach,
+        m_platform_info,
+        m_get_licenses,
+        m_cloud_type,
+        FakeConfig,
+    ):
+        m_cloud_type.return_value = ("gce", None)
+        m_get_licenses.side_effect = TypeError("error")
+        m_platform_info.return_value = {"series": "ubuntu-lts"}
+        cfg = FakeConfig()
+
+        assert gcp_auto_attach(cfg) is None
```

Comment:
Can we add `call_count` check for the `m_get_licenses` mock ?

Reply:
Yes sure